demisto_sdk/commands/validate/validators/PB_validators/PB121_is_correct_value_references_interface.py

```diff
@@ -0,0 +1,198 @@
+        ' Change the reference to "From previous tasks" from "As value", or change the value to ${{{path}}}.'
+    )
+    related_field = "conditions"
+    is_auto_fixable = True
+    fix_message = "Fixed the following inputs:\n"
+    handle_value_obj: Callable = lambda x: x
+
+    def obtain_invalid_content_items(
+        self, content_items: Iterable[ContentTypes]
+    ) -> list[ValidationResult]:
+        """Check that references of context values, are valid, i.e. "iscontext: true" or surrounded by ${<condition>},
+        Args:
+            content_items (Iterable[ContentTypes]): The content items to check.
+        Returns:
+            List[ValidationResult]. List of ValidationResults objects.
+        """
+        self.handle_value_obj = self.get_invalid_value_obj
+
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(
+                    task_id=task.id,
+                    task_name=task.task.name,
+                    path=value,
+                ),
+                content_object=playbook,
+            )
+            for playbook in content_items
+            for task, value in self.obtain_invalid_playbook_inputs(playbook)
+        ]
+
+    def obtain_invalid_playbook_inputs(
+        self, playbook: ContentTypes
+    ) -> list[tuple[TaskConfig, str]]:
+        results: list[tuple[TaskConfig, str]] = []
+
+        for task in playbook.tasks.values():
+            is_task_valid = {
+                "condition": self.is_valid_condition_task,
+                "regular": self.is_valid_regular_task,
+                "collection": self.is_valid_data_collection,
+            }.get(task.type, lambda _: [])  # type: ignore
+
+            invalid_values = (
+                is_task_valid(task)
+                + self.handle_value_obj(task.task, "description")
+                + self.handle_value_obj(task.task, "name")
+            )
+
+            results += [(task, val) for val in invalid_values]
+
+        return results
+
+    def is_valid_condition_task(self, task: TaskConfig) -> list[str]:
+        task.task.description
+        invalid_values = []
+        for conditions in task.conditions or []:
+            for condition in conditions.get("condition", []):
+                for condition_info in condition:
+                    invalid_values += self.handle_op_arg(
+                        **condition_info.get("left", {})
+                    ) + self.handle_op_arg(**condition_info.get("right", {}))
+        for message_key, message_value in (task.message or {}).items():
+            invalid_values += self.get_invalid_message_values(
+                message_key, message_value
+            )
+        for script_argument in (task.scriptarguments or {}).values():
+            invalid_values += self.handle_input_obj(script_argument)
+        return invalid_values
+
+    def is_valid_regular_task(self, task: TaskConfig) -> list[str]:
```

Comment:
let's rename to obtain_invalid regular_tasks or something similar,
makes more sense when when we're returning a list of invalid vlues.